primitives/src/assets.rs

```diff
@@ -471,4 +489,27 @@ where
 		}
 		.map_err(|e| FungibleLedgerError::InvalidTransfer(e))
 	}
+
```

Comment:
Actually the `transfer()` actually tries to keep accounts alive, instead of allowing death. 
This is inconsistent with the behavior of `withdraw` that we kept so far and that I kept with `AllowDeath` in the `burn` impl